operator/src/main/resources/scripts/startServer.sh

```diff
@@ -126,11 +147,93 @@ function copySitCfg() {
+
+  trace "Model-in-Image: Creating domain home."
+
+  # primordial domain contain the basic structures, security and other fmwconfig templated info
+  # domainzip only contains the domain configuration (config.xml jdbc/ jms/)
+  # Both are needed for the complete domain reconstruction
+
+  if [ ! -f /weblogic-operator/introspector/primordial_domainzip.secure ] ; then
+    trace "Domain Source Type is FromModel, the primordial model archive is missing, cannot start server"
+    exit 1
+  fi
+
+  if [ ! -f /weblogic-operator/introspector/domainzip.secure ] ; then
+    trace "Domain type is FromModel, the domain configuration archive is missing, cannot start server"
+    exit 1
+  fi
+
+  cd / && base64 -d /weblogic-operator/introspector/primordial_domainzip.secure > /tmp/domain.tar.gz && \
+   tar -xzf /tmp/domain.tar.gz
+
+  # decrypt the SerializedSystemIni first
+  if [ -f ${RUNTIME_ENCRYPTION_SECRET_PASSWORD} ] ; then
+    MII_PASSPHRASE=$(cat ${RUNTIME_ENCRYPTION_SECRET_PASSWORD})
+  else
+    trace SEVERE "Domain Source Type is 'FromModel' which requires specifying a runtimeEncryptionSecret " \
+    "in your domain resource and deploying this secret with a 'password' key, but the secret does not have this key."
+    exit 1
+  fi
+  encrypt_decrypt_domain_secret "decrypt" ${DOMAIN_HOME} ${MII_PASSPHRASE}
+
+  # restore the config zip
+  #
+  cd / && base64 -d /weblogic-operator/introspector/domainzip.secure > /tmp/domain.tar.gz && \
+    tar -xzf /tmp/domain.tar.gz
+  chmod +x ${DOMAIN_HOME}/bin/*.sh ${DOMAIN_HOME}/*.sh
+
+  # restore the archive apps and libraries
+  #
+  trace "Model-in-Image: Deploying libraries."
+
+  mkdir -p ${DOMAIN_HOME}/lib
+  if [ $? -ne 0 ] ; then
+    trace  SEVERE "Domain Source Type is FromModel, cannot create ${DOMAIN_HOME}/lib "
+    exit 1
+  fi
+
+  for file in $(sort_files ${IMG_ARCHIVES_ROOTDIR} "*.zip")
+    do
+        # expand the archive domain libraries to the domain lib
+        cd ${DOMAIN_HOME}/lib  || exit 1
+        ${JAVA_HOME}/bin/jar xf ${IMG_ARCHIVES_ROOTDIR}/${file} wlsdeploy/domainLibraries/
+
+        if [ $? -ne 0 ] ; then
+          trace SEVERE  "Domain Source Type is FromModel, error in extracting application archive ${IMG_ARCHIVES_ROOTDIR}/${file}"
+          exit 1
+        fi
+
+        # expand the archive apps and shared lib to the wlsdeploy/* directories
+        # the config.xml is referencing them from that path
+
+        cd ${DOMAIN_HOME} || exit 1
+        ${JAVA_HOME}/bin/jar xf ${IMG_ARCHIVES_ROOTDIR}/${file} wlsdeploy/
+
+        if [ $? -ne 0 ] ; then
+          trace SEVERE "Domain Source Type is FromModel, error in extracting application archive ${IMG_ARCHIVES_ROOTDIR}/${file}"
+          exit 1
+        fi
+    done
+
+}
+
 # trace env vars and dirs before export.*Home calls
 
 traceEnv before
 traceDirs before
 
+traceTiming "POD '${SERVICE_NAME}' MII UNZIP START"
+
+if [ -f /weblogic-operator/introspector/domainzip.secure ]; then
+  prepareMIIServer
+fi
```

Comment:
It looks like 'exportEffectiveDomainHome' further below may be called too late?  Doesn't the above depend on the domain home to be correct?    

The problem is that an 'empty' domain home may actually mean the domain home is not in the expected location - which is what export effective domain home corrects.   A bit of a chicken-and-the-egg problem, but it needs to be solved, as the 'default domain home' is actually a dynamic calculation.

Reply:
Ignore this.  Resolving.